cmd/argocd/commands/app.go

```diff
@@ -1099,7 +1101,12 @@ func findandPrintDiff(ctx context.Context, app *argoappv1.Application, proj *arg
 			if !foundDiffs {
 				foundDiffs = true
 			}
-			_ = cli.PrintDiff(item.key.Name, live, target)
+			err = cli.PrintDiff(item.key.Name, live, target)
+			// ignore exitError since diff utilities usually return exit status 1 when diff is present
+			var execError *exec.Error
+			if err != nil && goerrors.As(err, &execError) {
+				errors.CheckError(err)
```

Comment:
Btw, I noticed that the diff is also used in the `app sync` command [here](https://github.com/argoproj/argo-cd/blob/master/cmd/argocd/commands/app.go#L1776) and I assume failing fast would break a lot of CI jobs. Perhaps, just log with a warning that diff is missing because of so and so error?

Reply:
Good catch. What if we just let this function return the error up. `diff` can fail, `sync` can just log and move on.

Reply:
Hi, I've updated it to bubble up the error. Can you please take a look again?